notifier/arm64_build/build_arm

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+#CHange dir to the root of the project
+cd ..
+
+GOOS=linux
+GOARCH=arm64
+
+# get build data
+if [[ "$BUILD_COMMIT" == "" ]]; then
+  BUILD_COMMIT=$(git describe --all --long --abbrev=99 --dirty 2>/dev/null)
+fi
+if [[ "$BUILD_USER" == "" ]]; then
+  BUILD_USER=$(id -un)
+fi
+if [[ "$BUILD_HOST" == "" ]]; then
+  BUILD_HOST=$(hostname -f)
+fi
+if [[ "$BUILD_DATE" == "" ]]; then
+  BUILD_DATE=$(date +%d.%m.%Y)
+fi
+if [[ "$BUILD_SOURCE" == "" ]]; then
+  BUILD_SOURCE=$(git remote -v | grep origin | cut -f2 | cut -d" " -f1 | head -n 1)
+fi
+if [[ "$BUILD_SOURCE" == "" ]]; then
+  BUILD_SOURCE=$(git remote -v | cut -f2 | cut -d" " -f1 | head -n 1)
+fi
+BUILD_BUILDOPTIONS=$(echo $* | sed "s/ /§/g")
+
+# check
+if [[ "$BUILD_COMMIT" == "" ]]; then
+  echo "could not automatically determine BUILD_COMMIT, please supply manually as environment variable."
+  exit 1
+fi
+if [[ "$BUILD_USER" == "" ]]; then
+  echo "could not automatically determine BUILD_USER, please supply manually as environment variable."
+  exit 1
+fi
+if [[ "$BUILD_HOST" == "" ]]; then
+  echo "could not automatically determine BUILD_HOST, please supply manually as environment variable."
+  exit 1
+fi
+if [[ "$BUILD_DATE" == "" ]]; then
+  echo "could not automatically determine BUILD_DATE, please supply manually as environment variable."
+  exit 1
+fi
+if [[ "$BUILD_SOURCE" == "" ]]; then
+  echo "could not automatically determine BUILD_SOURCE, please supply manually as environment variable."
+  exit 1
+fi
+
+echo "Please notice, that this build script includes metadata into the build."
+echo "This information is useful for debugging and license compliance."
+echo "Run the compiled binary with the -version flag to see the information included."
+
+# build
+BUILD_PATH="github.com/safing/portbase/info"
+go build -ldflags "-X ${BUILD_PATH}.commit=${BUILD_COMMIT} -X ${BUILD_PATH}.buildOptions=${BUILD_BUILDOPTIONS} -X ${BUILD_PATH}.buildUser=${BUILD_USER} -X ${BUILD_PATH}.buildHost=${BUILD_HOST} -X ${BUILD_PATH}.buildDate=${BUILD_DATE} -X ${BUILD_PATH}.buildSource=${BUILD_SOURCE}" -o arm64_build/notifier_arm $*
+
+go build -ldflags "-X ${BUILD_PATH}.commit=${BUILD_COMMIT} -X ${BUILD_PATH}.buildOptions=${BUILD_BUILDOPTIONS} -X ${BUILD_PATH}.buildUser=${BUILD_USER} -X ${BUILD_PATH}.buildHost=${BUILD_HOST} -X ${BUILD_PATH}.buildDate=${BUILD_DATE} -X ${BUILD_PATH}.buildSource=${BUILD_SOURCE}" -tags=legacy_appindicator -o arm64_build/notifier_arm_legacy $*
```

Comment:
Ideally, the build process should fit into our current build process, which puts all assets in the "dist" directory, which lives at the root of the repo and is symlinked with other repos.

The pack script takes the output of the build script and marks it with correct build os, arch and version and places it in the correct directory.

Building is also done from a single container (except for electron), which has cross compiling support, configured as seen in https://github.com/safing/portmaster/blob/0c5bdbbc1302d21d109abb74be0a8701399f333b/cmds/portmaster-start/build#L51-L63.

(We should probably start publishing our building container - I've been thinking about creating a builder image that will download all repos and do all builds in one go.)

I haven't had time yet to test the PR, but from a first look, I don't see the tag `legacy_appindicator` being used.

For a deeper handling of this, I'll need to set aside some time next sprint.

Thanks again for the PR!

Reply:
The arm build for the notifier is going to be different from the rest since it is using the c library.
I was thinking how to integrate it in to the current build process but the only way Im familiar is with running some kind of virtualization (assuming your trying to build it with x86 host architecture) so I decided to make it independent for now.
Note that docker needs to be configured in order to run the docker image in the arm build you can check the readme.

I'm not sure what you mean by that the `legacy_appindicator` tag is not used.

I will be happy to help if there is proper solution to the build process.
Thanks for the quick response.